ee/query-service/app/api/pat.go

```diff
@@ -0,0 +1,105 @@
+package api
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"go.signoz.io/signoz/ee/query-service/model"
+	"go.signoz.io/signoz/pkg/query-service/auth"
+	"go.uber.org/zap"
+)
+
+func generatePATToken() string {
+	// Generate a 32-byte random token.
+	token := make([]byte, 32)
+	rand.Read(token)
+	// Encode the token in base64.
+	encodedToken := base64.StdEncoding.EncodeToString(token)
+	return encodedToken
+}
+
+func (ah *APIHandler) createPAT(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	req := model.PAT{}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		RespondError(w, model.BadRequest(err), nil)
+		return
+	}
+	user, err := auth.GetUserFromRequest(r)
+	if err != nil {
+		RespondError(w, &model.ApiError{
+			Typ: model.ErrorUnauthorized,
+			Err: err,
+		}, nil)
+		return
+	}
+
+	req.UserID = user.Id
```

Comment:
can we add a comment to mention that all pats will be assigned to user sending the request for now. But in future this may change. 

You could also create a struct PostablePAT{} that has only supported fields from clients (browser). This means PostablePAT{} will not have userId, token but other fields. 

PostablePAT{
	Name      string `json:"name" db:"name"`
}

PAT {
        PostablePAT
        Id        string `json:"id" db:"id"`
	UserID    string `json:"userId" db:"user_id"`
	Token     string `json:"token" db:"token"`
	CreatedAt int64  `json:"createdAt" db:"created_at"`
	ExpiresAt int64  `json:"expiresAt" db:"expires_at"`
}

this pattern is more of personal choice though, so its your call if you think its worth doing

Reply:
Added the comment.

We can create PostablePAT, but I think it makes sense to return the token in the createPAT API call.
I'll create a masked PAT (using PostablePAT) for all other use cases.